### PR TITLE
Implement additional unreachable cases for break, continue and raise

### PR DESCRIPTION
**Pull request by RJ722**
Closes: #84 

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation in the README file.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: vulture/core.py [hunk 1]
```diff
@@ -425,14 +425,22 @@ def _handle_ast_list(self, ast_list):
...
                 try:
                     first_unreachable_node = ast_list[index + 1]
                 except IndexError:
                     continue
+                node_names = {
+                    ast.Break: 'break',
+                    ast.Continue: 'continue',
+                    ast.Raise: 'raise',
+                    ast.Return: 'return'
+                }
```
Comment (RJ722) [inline vulture/core.py]: Should this be moved out of the function, with other constants?

File: tests/test_unreachable.py [hunk 2]
```diff
@@ -104,3 +104,223 @@ def foo():
         return False
 """)
     check_unreachable(v, 4, 1, 'return')
+
+
+def test_raise_assignemnt(v):
```
Comment (jendrikseipp) [inline tests/test_unreachable.py]: typo

File: tests/test_unreachable.py [hunk 3]
```diff
@@ -104,3 +104,223 @@ def foo():
...
+""")
+    check_unreachable(v, 6, 1, 'continue')
+
+
+def test_continue_one_liner(v):
+    v.scan("""\
+def foo():
+    for i in range(1, 10):
+        if i == 5: continue
+        print(1/i)
```
Comment (jendrikseipp) [inline tests/test_unreachable.py]: `1 / i`

File: tests/test_unreachable.py [hunk 4]
```diff
@@ -104,3 +104,223 @@ def foo():
...
+""")
+    assert v.unreachable_code == []
+
+
+def test_continue_with_comment_only(v):
```
Comment (jendrikseipp) [inline tests/test_unreachable.py]: I don't think we need the extra tests that end with comments. For the AST the comments don't even exist.

File: vulture/core.py [hunk 5]
```diff
@@ -425,14 +425,22 @@ def _handle_ast_list(self, ast_list):
         Find unreachable nodes in the given sequence of ast nodes.
         """
         for index, node in enumerate(ast_list):
-            if isinstance(node, ast.Return):
+            nodes_triggering_unreachable = (ast.Break, ast.Continue, ast.Raise,
```
Comment (jendrikseipp) [inline vulture/core.py]: no need for the extra variable, just put the tuple below.

File: vulture/core.py [hunk 6]
```diff
@@ -425,14 +425,22 @@ def _handle_ast_list(self, ast_list):
...
+                node_names = {
+                    ast.Break: 'break',
+                    ast.Continue: 'continue',
+                    ast.Raise: 'raise',
+                    ast.Return: 'return'
+                }
                 self._define(
                     self.unreachable_code,
-                    'return',
+                    node_names[type(node)],
```
Comment (jendrikseipp) [inline vulture/core.py]: no need for the extra dict, use `node.__class__.__name__.lower()`

Comment (RJ722) [inline vulture/core.py]: Thanks! :)

**Comment by RJ722** [comment]
Thanks @jendrikseipp :-)

**Comment by jendrikseipp** [comment]
Thanks!

**Pull request merged**